### ISSUE TITLE
Bug with JS autocompletion

### ISSUE THREAD

**Issue by Tchoupinax** [adobe/brackets#12197]
Hello,

i found a bug about JS autocompletion on Brackets.
Let's create an html file like this : 

`<html>
<head>
    <title>Home</title>
    <meta charset="utf-8">
    <script>
    </script>
    <style>
    </style>
</head>
<body>
</body>
</html>`

Then, all is working right. When you write "d" between script tag, Brackets gives you some suggestion. Now, insert in body this code `"<p>...</p>"`. You can replace p by other tag like h1, h3, etc...
After that, save your file and restart brackets (Close it and start it). Then, after restarting it, try to write again "d" between script tag. Autocompletion will not work.

While you do not restart Brackets, this code `"<p>...</p>"` prevent JS autocompletion from working. 
IMPORTANT. If you close Brackets with this file overview, this bug will appear. However, if you close it with an other file, autocompletion will work perfectly while do not open this file in brackets.

Autocompletion works for CSS or HTML anyway. 

Sorry for my bad english. If you need more information on what i did, just ask it.

Regards,

**Comment by petetnt** [comment]
Hi @Tchoupinax, can confirm!

Actually just `..` is enough: 

```
<html> 
	<head>
		<title>Home</title> 
		<meta charset="utf-8"> 
		<script>d</script> 
		<style> </style> 
	</head> 
	<body>..</body> 
</html>
```
This will throw the following: 

```
 Worker: Error thrown in tern_worker:undefined is not a function
TypeError: undefined is not a function
    at infer (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:1100:39)
    at file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:987:9
    at Object.BinaryExpression (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:906:15)
    at infer (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:1100:39)
    at file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:986:9
    at Object.BinaryExpression (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:906:15)
    at infer (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:1100:39)
    at Object.walk.make.Expression (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/tern/lib/infer.js:1105:7)
    at c (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/acorn/util/walk.js:60:37)
    at Object.base.ExpressionStatement (file:///C:/Program%20Files%20(x86)/Brackets/dev/src/extensions/default/JavaScriptCodeHints/thirdparty/acorn/util/walk.js:176:5)
```

TernWorker will fail to walk over the `..` item because you cannot call methods `.`

**Comment by K-Thompson** [comment]
@petetnt 

See:

[Error raised in tern_worker when parsing HTML #11162](https://github.com/adobe/brackets/issues/11162)

Perhaps some research has already been done?


Further, personally, code hinting and autocomplete are main reasons for using Brackets. But, I think hinting and auto complete is unreliable and these issues need to be more carefully track. 

See [Code Hints Unreliable #11570](https://github.com/adobe/brackets/issues/11570)

[Code Hints Don't work for me any longer (JS only) #12322 (https://github.com/adobe/brackets/issues/12322)

**Comment by petetnt** [comment]
@K-Thompson have you tested the code hints against newer versions of Acorn / Tern (such as with #11569?)

**Comment by MarcelGerber** [comment]
This is fixed with the merge of #11569 and will come to you in Release 1.8.
Please let us know if it still doesn't work for you when the Release ships.
